### PR TITLE
Fix bug: Use try-with-resources or close this "Socket" in a "finally" clause. 

### PR DESCRIPTION
**Pull request by Jave-Chen**
## What is the purpose of the pull request
#1714 
Fix bug: Use try-with-resources or close this "Socket" in a "finally" clause. 

## Brief change log
1. Use try-with-resources with Socket, prepareStatement.
2. Add connection.close() at executeFuncAndSql.
3. Fix some code smell.

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void handle() throws Exception {
         // set the name of the current thread
         String threadLoggerInfoName = String.format(Constants.TASK_LOG_INFO_FORMAT, taskProps.getTaskAppId());
         Thread.currentThread().setName(threadLoggerInfoName);
-        logger.info(sqlParameters.toString());
+        logger.info("{}", sqlParameters);
```
[the remark this answers is not in the document]
Comment (Technoboy-) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java]: could we add more prefix message ,like :
```
logger.info("sqlParameters {}", sqlParameters);
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/FourLetterWordMain.java [hunk 2]
```diff
@@ -59,23 +59,22 @@ public static String send4LetterWord(String host, int port, String cmd)
      */
     public static String send4LetterWord(String host, int port, String cmd, int timeout)
             throws IOException {
-        LOG.info("connecting to " + host + " " + port);
-        Socket sock = new Socket();
+        LOG.info("connecting to {0} {1}", host, port);
```
Comment (Technoboy-) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/FourLetterWordMain.java]: why {0}, {1} ?

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jave-Chen) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java]: SqlParameters override toString method.

[the remark this answers is not in the document]
Comment (Jave-Chen) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java]: SqlParameters override toString method.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/FourLetterWordMain.java (hunk 2 above)
Comment (Jave-Chen) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/FourLetterWordMain.java]: fixed

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jave-Chen) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java]: > could we add more prefix message ,like :
> 
> ```
> logger.info("sqlParameters {}", sqlParameters);
> ```

done

**Comment by dailidong** [comment]
+1

**Pull request merged**